# KUNCI JAWABAN MODUL 2

## Built-in Function Input

In [1]:
nama = input('Nama : ')
nim = input('NIM : ')
umur = input('Umur : ')
fakultas = input('Fakultas : ')
prodi = input('Program Studi : ')

print ('Mahasiswa bernama {a} berumur {c} tahun. {a} adalah mahasiswa {d} dan mengambil program studi {e}. NIM {a} adalah {b}.'.format(
    a = nama, b = nim, c= umur, d=fakultas, e=prodi))

Nama : Rini
NIM : 02030001
Umur : 21
Fakultas : FaSt
Program Studi : Elektro
Mahasiswa bernama Rini berumur 21 tahun. Rini adalah mahasiswa FaSt dan mengambil program studi Elektro. NIM Rini adalah 02030001.


## Nested-If

In [2]:
a = int(input ('Nilai a = '))
b = int(input ('Nilai b = '))
c = int(input ('Nilai c = '))

if (a > 10) :
    if (b >= 5) :
        print ('V')
    else :
        print ('W')
elif (a < 10) :
    if (c > 7) :
        print ('X')
    elif (c <= 7) :
        print('Y')
else :
    print ('Z')

Nilai a = 21
Nilai b = 3
Nilai c = 10
W


## Nested Loop

In [3]:
panjang = int(input('panjang persegi : '))
lebar = int(input ('lebar persegi : '))

for i in range(lebar):
    for j in range (panjang):
        print('*  ', end = '')
    print('')

panjang persegi : 3
lebar persegi : 3
*  *  *  
*  *  *  
*  *  *  


## Function, Mapping, Lambda Expression

In [4]:
def square_maker ():
    panjang = int(input('panjang persegi : '))
    lebar = int(input ('lebar persegi : '))

    for i in range(lebar):
        for j in range (panjang):
            print('*  ', end = '')
        print('')

In [5]:
square_maker()

panjang persegi : 4
lebar persegi : 4
*  *  *  *  
*  *  *  *  
*  *  *  *  
*  *  *  *  


In [6]:
listofnum = [1,2,3,4,5,6]
list(map (lambda x: ((x)**3)*4*3.14/3, listofnum))

[4.1866666666666665,
 33.49333333333333,
 113.04,
 267.94666666666666,
 523.3333333333334,
 904.32]

# Pertanyaan

## 1.	Pada program A, jika pada function input NIM dimasukkan 02030001 lalu dikonversi menjadi suatu integer, apa hasil print yang dikeluarkan?

##### Hasil yang keluar adalah 2030001, karena integer tidak menghitung '0' pada bagian awal.

In [7]:
nama = input('Nama : ')
nim = int(input('NIM : '))
umur = input('Umur : ')
fakultas = input('Fakultas : ')
prodi = input('Program Studi : ')

print ('Mahasiswa bernama {a} berumur {c} tahun. {a} adalah mahasiswa {d} dan mengambil program studi {e}. NIM {a} adalah {b}.'.format(
    a = nama, b = nim, c= umur, d=fakultas, e=prodi))

Nama : Rini
NIM : 02030001
Umur : 21
Fakultas : FaST
Program Studi : Elektro
Mahasiswa bernama Rini berumur 21 tahun. Rini adalah mahasiswa FaST dan mengambil program studi Elektro. NIM Rini adalah 2030001.


## 2.	Bagaimana cara agar keluaran program A menghasilkan kalimat yang seluruhnya berhuruf kapital?  

##### Gunakan function .upper() untuk mengubah setiap elemen dalam string menjadi uppercase

In [8]:
nama = input('Nama : ')
nim = int(input('NIM : '))
umur = input('Umur : ')
fakultas = input('Fakultas : ')
prodi = input('Program Studi : ')

print ('Mahasiswa bernama {a} berumur {c} tahun. {a} adalah mahasiswa {d} dan mengambil program studi {e}. NIM {a} adalah {b}.'.format(
    a = nama, b = nim, c= umur, d=fakultas, e=prodi).upper())

Nama : Rini
NIM : 02030001
Umur : 21
Fakultas : FaST
Program Studi : Elektro
MAHASISWA BERNAMA RINI BERUMUR 21 TAHUN. RINI ADALAH MAHASISWA FAST DAN MENGAMBIL PROGRAM STUDI ELEKTRO. NIM RINI ADALAH 2030001.


## 3.	Bagaimana cara agar keluaran program A menghasilkan kalimat tanpa ada huruf vokal?  Buatlah program yang dimodifikasi tersebut pada file notebook yang sama!

In [9]:
nama = input('Nama : ')
nim = int(input('NIM : '))
umur = input('Umur : ')
fakultas = input('Fakultas : ')
prodi = input('Program Studi : ')

string = 'Mahasiswa bernama {a} berumur {c} tahun. {a} adalah mahasiswa {d} dan mengambil program studi {e}. NIM {a} adalah {b}.'.format(
    a = nama, b = nim, c= umur, d=fakultas, e=prodi)
new_string=''
vowels = ('a', 'i', 'u', 'e', 'o', 'A', 'I', 'U', 'E', 'O')
for i in string:
    if i in vowels:
        a=1
    else:
        new_string = new_string+i

print(new_string)

Nama : Rini
NIM : 02030001
Umur : 21
Fakultas : FaST
Program Studi : Elektro
Mhssw brnm Rn brmr 21 thn. Rn dlh mhssw FST dn mngmbl prgrm std lktr. NM Rn dlh 2030001.


## 4. Apa keluaran program pada program B jika user memasukkan input dengan skenario sebagai berikut?

In [10]:
for i in range (12):
    a = int(input ('Nilai a = '))
    b = int(input ('Nilai b = '))
    c = int(input ('Nilai c = '))

    if (a > 10) :
        if (b >= 5) :
            print ('V')
        else :
            print ('W')
    elif (a < 10) :
        if (c > 7) :
            print ('X')
        elif (c <= 7) :
            print('Y')
    else :
        print ('Z')

Nilai a = 20
Nilai b = 3
Nilai c = 7
W
Nilai a = 15
Nilai b = 2
Nilai c = 8
W
Nilai a = 9
Nilai b = 2
Nilai c = 4
Y
Nilai a = 14
Nilai b = 15
Nilai c = 3
V
Nilai a = 1
Nilai b = 7
Nilai c = 10
X
Nilai a = 10
Nilai b = 1
Nilai c = 6
Z
Nilai a = 2
Nilai b = 3
Nilai c = 1
Y
Nilai a = 11
Nilai b = 5
Nilai c = 8
V
Nilai a = 4
Nilai b = 6
Nilai c = 7
Y
Nilai a = 9
Nilai b = 1
Nilai c = 9
X
Nilai a = 10
Nilai b = 9
Nilai c = 9
Z
Nilai a = 3
Nilai b = 7
Nilai c = 4
Y


## 5.	Buatlah program untuk menampilkan bentuk jajaran genjang. User akan memasukkan dua variabel yaitu panjang dan tinggi.
Keluaran seperti apa yang akan muncul jika user memasukan angka 2 dan 4? 

In [11]:
panjang= int(input('panjang : '))
tinggi = int(input ('tinggi : '))
space = 1
for i in range (tinggi):
    for j in range (panjang):
        print ('*  ', end='')
    print ('')
    for k in range (space):
        print ('  ', end='')
    space = space +1

panjang : 2
tinggi : 4
*  *  
  *  *  
    *  *  
      *  *  
        

## 6.	Apa perbedaan antara function dan lambda expression?	 
Buatlah suatu readme file pada folder Modul 2 GitHub pribadi anda dan jelaskan apa yang membedakan antara function dan lambda expression. 
Buatlah penjelasan dengan menyertakan gambar contoh function yang disederhanakan menjadi suatu lambda expression.


##### Function adalah sekumpulan code yang dapat digunakan berkali - kali

##### Lambda Expression adalah function yang diringkas menjadi function anonimus

In [12]:
def square (x):
    a = x*x
    return a

In [13]:
def square (x):
    return (x*x)

In [14]:
lambda x: x*x

<function __main__.<lambda>(x)>

## 7.	Apa  perbedaan antara function, method, dan property? Jelaskan perbedaan antara ketiganya pada file readme yang sama!

Function : sekumpulan kode independen yang bisa dipanggil berkali - kali sehingga lebih praktis

Method : function yang terhubung dengan suatu class pada object oriented programming (OOP)


Properties : sebutan pada OOP yang berfungsi untuk mengambil value atau mengatur value